### PR TITLE
PAYARA-3090 Support BASIC authentication with additional micro repositories

### PR DESCRIPTION
**Pull request by Cousjava**

### CODE REVIEW

**Comment by Cousjava** [comment]
Jenkins test please

**Comment by Cousjava** [comment]
Jenkins test please

File: nucleus/deployment/common/src/main/java/fish/payara/deployment/util/GAVConvertor.java [hunk 1]
```diff
@@ -80,7 +84,29 @@
...
+     * @param GAV A comma separated list of the groupId, artifactId, 
+     * and version number
+     * @param repositoryURLs A collection of repository URLs to search for the 
+     * target artefact in
+     * @return A URL matching the provided GAV
+     * @throws MalformedURLException Thrown if an artefact cannot be found for
+     * the provided GAV
+     */
+    public Map.Entry<String, URL> getArtefactMapEntry(String GAV, Collection<String> repositoryURLs) throws MalformedURLException {
+        List<URL> repoURLs = new LinkedList<URL>();
```
Comment (rdebusscher) [inline nucleus/deployment/common/src/main/java/fish/payara/deployment/util/GAVConvertor.java]: No type needed on LinkedList

File: appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java [hunk 2]
```diff
@@ -667,15 +667,7 @@ public PayaraMicroImpl addRepoUrl(String... URLs) {
...
-            try {
-                if (!url.endsWith("/")) {
-                    repositoryURLs.add(new URL(url + "/"));
-                } else {
-                    repositoryURLs.add(new URL(url));
-                }
-            } catch (MalformedURLException ex) {
-                LOGGER.log(Level.SEVERE, "{0} is not a valid URL and will be ignored", url);
-            }
+            repositoryURLs.add(url);
```
Comment (rdebusscher) [inline appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java]: for loop can be replaced with Collections.addAll(repositoryURLs, URLs);

**Comment by Cousjava** [comment]
Jenkins test please

**Pull request merged**